src/widgets/Tabs/PatientSelect.js

```diff
@@ -54,22 +61,78 @@ import { Paper } from '../Paper';
  * @prop {String} sortKey               Current key the list of patients is sorted by.
  *
  */
+
+const getMessage = (noResults, error) => {
+  if (noResults) return generalStrings.could_not_find_patient;
+  if (error) return generalStrings.error_communicating_with_server;
+  return generalStrings.enter_patient_details;
+};
+
+const EmptyComponent = ({ loading, error, searchedWithNoResults }) => (
+  <FlexView flex={1} justifyContent="center" alignItems="center" style={{ marginTop: 20 }}>
+    {loading ? (
+      <ActivityIndicator color={SUSSOL_ORANGE} size="small" />
+    ) : (
+      <Text style={{ fontFamily: APP_FONT_FAMILY, fontSize: APP_GENERAL_FONT_SIZE }}>
+        {getMessage(searchedWithNoResults, error)}
+      </Text>
+    )}
+  </FlexView>
+);
+
+const Header = ({ onSearchOnline, onNewPatient }) => (
+  <FlexRow justifyContent="center" alignItems="center">
+    <Text
+      style={{
+        fontFamily: APP_FONT_FAMILY,
+        color: DARKER_GREY,
+        fontSize: 14,
+      }}
+    >
+      {vaccineStrings.vaccine_dispense_step_one_title}
+    </Text>
+    <View style={{ flex: 1, marginLeft: 'auto' }} />
+    <PageButton style={{ height: 10 }} text="Search online" onPress={onSearchOnline} />
+    <PageButton
+      style={{ height: 10, marginLeft: 10 }}
+      text={`${dispensingStrings.new} ${dispensingStrings.patient}`}
+      onPress={onNewPatient}
+    />
+  </FlexRow>
+);
+
+Header.propTypes = {
+  onSearchOnline: PropTypes.func.isRequired,
+  onNewPatient: PropTypes.func.isRequired,
+};
+
+EmptyComponent.propTypes = {
+  loading: PropTypes.bool.isRequired,
+  error: PropTypes.oneOfType([PropTypes.object, PropTypes.bool]).isRequired,
+  searchedWithNoResults: PropTypes.bool.isRequired,
+};
+
 const PatientSelectComponent = ({
   createPatient,
   formConfig,
   onCancelPrescription,
-  onFilterData,
-  patients,
   selectPatient,
   updateForm,
+  completedForm,
 }) => {
+  const [
+    { data, loading, searchedWithNoResults, error },
+    onSearchOnline,
+    filter,
+  ] = useLocalAndRemotePatients([]);
+
   const columns = React.useMemo(() => getColumns(MODALS.PATIENT_LOOKUP), []);
   const { pageTopViewContainer } = globalStyles;
   const keyboardIsOpen = useKeyboardIsOpen();
-  const debouncedFilter = useDebounce(onFilterData, 300);
+
   const handleUpdate = (key, value) => {
     updateForm(key, value);
-    debouncedFilter(key, value);
+    filter({ ...completedForm, [key]: value });
```

Comment:
need to do this here as completedForm is not correct until the above `updateForm` has been dispatched